rust/src/fvm/machine.rs

```diff
@@ -132,6 +132,84 @@ fn create_fvm_machine(
     }
 }
 
+#[ffi_export]
+fn create_fvm_debug_machine(
```

Comment:
Can we refactor this to avoid duplicating the whole function?

Reply:
sure, once it provably works :)

Reply:
done.